OpenSaveDlg.cpp

```diff
@@ -594,7 +617,20 @@ UINT_PTR CALLBACK SaveFileDlg::OfnHook( HWND dlg, UINT msg, WPARAM wp, LPARAM lp
 			for( ulong i=0; i<csl.size(); ++i )
 				if( csl[i].type & 1 ) // 1:=SAVE
 					cb.Add( csl[i].longName );
-			cb.Select( csl[pThis->csIndex_].longName );
```

Comment:
@roytam1 this cls[pThis->csIndex_] index range was not checked (ans is not checked on your branch) This might lead to a crash when opening the **Save As...** dialog if the current CP is not in the list (this can happen also with your build).
I am trying to allow the user to always type cp number in the controls, in your case you would probably only be interested by some of the changes made to `SaveFileDlg::OfnHook()`